functions/src/db/getUserToken.ts

```diff
@@ -0,0 +1,26 @@
+import { UID } from '../types/uid'
+import { JWTToken } from '../types/JWT'
+import { db } from '../firebase/firebase'
+import { UserData } from '../types/UserData'
+
+export const getUserToken = async (userId: UID): Promise<JWTToken | null> => {
+    const userDocumentSnapshot = await db.collection('users').doc(userId).get()
+
+    if (!userDocumentSnapshot?.exists) {
+        return null
+    }
+
+    const userData = <UserData>userDocumentSnapshot.data()
+
+    if (userData.tokens) {
+        const validTokens = Object.keys(userData.tokens).filter(
+            (token) => userData.tokens[token].valid
+        )
+
+        if (validTokens.length > 0) {
+            return validTokens[0]
+        }
+    }
+
+    return null
```

Comment:
this as well seems to me a *No valid token found* error

Reply:
if this correctly returns `null`, then `getToken` as well would send back `null` to the client side, I guess